Add maximum radial distance feature

adrien-berchet opened the pull request:
Add a neurite feature to get the maximum radial distance from the soma of a collection of neurites.

arnaudon commented:
@adrien-berchet , in the end, we don't need that, right? I guess we can close it?

adrien-berchet commented:
Indeed we don't use it for now. But I am wondering if it is better to close or to merge (after rebase)? Someone might need it someday.

arnaudon commented:
ah yes, we could merge, if @wizmer has nothing against it!

wizmer commented:
We can merge it but you'd need to pass it as a PR (no longer a draft) and it needs a rebase

adrien-berchet commented:
@wizmer I had to update `neurom.features._get_feature_value_and_func` to make it work with your new registering system but I am not sure how you intended to deal with features with `shape=()`, can you check it plz?

arnaudon commented:
So `_get_feature_value_and_func` was not working previously with `shape=()` types of features?

adrien-berchet commented:
Exactly, because `neurom/features/__init__.py::91` is trying to build a list from a scalar instead of an iterable.

arnaudon commented:
So is this a real bug in neurom? The test were still working?

adrien-berchet commented:
The tests zere zorking because the features with `shape=()` were actually not registered because they were filtered here: `neurom/features/__init__.py::177`. But I don't know the reason of this filter.

arnaudon commented:
@asanin-epfl , could you merge that? Appart from adding a new feature, it fixes a small bug, see adrien last comment. Thank you!

arnaudon commented:
@adrien-berchet , can you rebase so we can merge to master?

asanin-epfl commented:
I will review after the rebase.

adrien-berchet commented:
> I will review after the rebase.

Done

asanin-epfl commented on neurom/features/__init__.py:
```diff
@@ -88,7 +88,13 @@ def _get_feature_value_and_func(feature_name, obj, **kwargs):
     feat = _find_feature_func(feature_name)
 
-    return np.array(list(feat(obj, **kwargs))), feat
+    res = feat(obj, **kwargs)
```
Why this change? Tests still pass with the old code.

adrien-berchet replied:
See my two previous comments:
https://github.com/BlueBrain/NeuroM/pull/849#issuecomment-787809109
https://github.com/BlueBrain/NeuroM/pull/849#issuecomment-787846393

And there is a TODO at ``neurom/features/__init__.py::176`` so I guess this change was more or less planned.

adrien-berchet commented:
I added a more usual feature so it makes it usable and points out the issue before these changes: it is not possible to use ``get_feature()`` for functions with ``shape=()``. So it is not possible to call it to just get a scalar, we always get a list of at least 1 element (I updated the behavior in the last commit).

asanin-epfl commented:
Ok, good to go then. Thank you!

adrien-berchet commented:
Thanks!

arnaudon commented:
Sorry, I'm late to the party: this is fixed here: https://github.com/BlueBrain/NeuroM/pull/884
